test/mocks/Wallet.ts

```diff
@@ -0,0 +1,31 @@
+/*
```

Comment:
Have you tried to use ethereum-waffle? https://ethereum-waffle.readthedocs.io/en/latest/index.html

Reply:
No. I'll try it tomorrow.

Reply:
Don't think it'll work. We need to mock the Provider of the Wallet. Waffle mocks the Wallet of the Provider.